pew/pew.py

```diff
@@ -373,11 +373,23 @@ def setvirtualenvproject(env, project):
 
 
 def setproject_cmd():
-    """Given a virtualenv directory and a project directory, set the virtualenv up to be associated with the project."""
-    env = os.environ.get('VIRTUAL_ENV', args.get(1))
-    project = args.get(2, os.path.abspath('.'))
-    if not env:
+    """Given a virtualenv directory and a project directory, set the
+    virtualenv up to be associated with the project."""
+    if len(args) == 4:
+        env = args.get(2)
+        project = args.get(3)
```

Comment:
I've only skimmed the code, but I'm not sure about these changes...

there wasn't a `args.get(3)` in the original code, and this behaves differently is invoked inside a virtualenv with

`pew setproject project_dir dummy_arg`

otoh, with the changed code this would give an error (which is better than silently ignoring the dummy_arg)

I don't like the imperative code, but now that the branches have a guaranteed length, it's best to just use `sys.argv` instead of the `args` dict. That is then used only in `toggleglobalsitepackages`, which should/could be modified to avoid relying on it.

The args dict was used just to ease writing a trivially simple parsing code, but the better approach would probably be to just use a parsing library (argparser doesn't cut it here imho... I'd like something like [optparse-applicative](https://github.com/pcapriotti/optparse-applicative))